mlrun/data_types/infer.py

```diff
@@ -38,7 +40,12 @@ def infer_schema_from_df(
 
     def upsert_entity(name, value_type):
         if name in current_entities:
-            entities[name].value_type = value_type
+            old_type = entities[name].value_type
+            if old_type != value_type:
+                logger.warning(
+                    f"Overriding type of entity '{name}' from '{old_type}' to '{value_type}'"
```

Comment:
```suggestion
                    f"Overriding type of entity '{name}' from '{old_type}' to '{value_type}'. This may result in unusable or inaccurate data."
```
Since we don't make any effort to recover from type mismatch at the moment, it makes sense to alert of the consequences.